web/src/components/atoms/DownloadButton/index.tsx

```diff
@@ -5,37 +5,69 @@ import Button, { ButtonProps } from "@reearth-cms/components/atoms/Button";
 import Icon from "@reearth-cms/components/atoms/Icon";
 import { useT } from "@reearth-cms/i18n";
 
+export type PreviewType = "GEO" | "GEO3D" | "IMAGE" | "MODEL3D" | "UNKNOWN";
+
+export type AssetFile = {
+  children?: AssetFile[];
+  contentType?: string;
+  name: string;
+  path: string;
+  size: number;
+};
+
+export type Asset = {
+  id: string;
+  createdAt: string;
+  createdBy: string;
+  createdByType: string;
+  file: AssetFile;
+  fileName: string;
+  previewType?: PreviewType;
+  projectId: string;
+  size: number;
+  url: string;
+  threadId: string;
+  comments: Comment[];
+};
+
+export type Comment = {
+  id: string;
+  author: string;
+  content: string;
+  createdAt: string;
+};
+
 type DownloadButtonProps = {
   title?: string;
-  filename: string;
-  url: string;
+  selected?: Asset[];
   displayDefaultIcon?: boolean;
 } & ButtonProps;
 
 const DownloadButton: React.FC<DownloadButtonProps> = ({
   title,
-  filename,
-  url,
+  selected,
   displayDefaultIcon,
   ...props
 }) => {
   const t = useT();
   const handleDownload: MouseEventHandler<HTMLElement> | undefined = useCallback(async () => {
-    const res = await fetch(url, {
-      method: "GET",
-      headers: {
-        "Access-Control-Allow-Origin": "*",
-      },
+    selected?.map(async s => {
+      const res = await fetch(s.url, {
+        method: "GET",
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+        },
+      });
+      const blob = await res.blob();
+      fileDownload(blob, s.fileName);
     });
-    const blob = await res.blob();
-    fileDownload(blob, filename);
-  }, [filename, url]);
+  }, [selected]);
 
   return (
     <Button
       icon={displayDefaultIcon && <Icon icon="download" />}
       onClick={handleDownload}
-      disabled={!url}
+      disabled={!selected}
```

Comment:
What if we passed an empty array as `selected` (the condition will give false and the button will be enabled), this will fail I think.

Reply:
Changed to `!selected || selected.length <= 0` to account for empty arrays (though where this component is being used sets selected to undefined